### PR TITLE
Add tools to create pre-built binary and upload

### PR DESCRIPTION
**Pull request by jam7**
This PR adds tools which I am using to build pre-built binary and upload them.
This PR also moves several tools from top directory to under tools directory.

### CODE REVIEW

File: tools/Makefile [hunk 1]
```diff
@@ -0,0 +1,51 @@
+CREW_PREFIX = /usr/local
+CREW_LIB_PATH = $(CREW_PREFIX)/lib/crew
```
Comment (uberhacker) [inline tools/Makefile]: Should this take into account the `/usr/local/lib64` directory on x86_64 architecture?

File: tools/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+case $1 in
+-k) keep="keep"; shift;;
+*)  keep="";;
+esac
+set -o pipefail
+# yes | crew install buildessential $keep
+arch=`uname -m`
+for i in "$@"; do
+    version=`grep '\<version ['"'"'"]' /usr/local/lib/crew/packages/$i.rb | head -1 | sed -e 's/^.*version *['"'"'"]//' -e 's/['"'"'"].*$//'`
```
Comment (uberhacker) [inline tools/build.sh]: Should we use `CREW_LIB_PREFIX` constant?

File: tools/genpkgname.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+version=`grep '\<version ['"'"'"]' /usr/local/lib/crew/packages/$1.rb | head -1 | sed -e 's/^.*version *['"'"'"]//' -e 's/['"'"'"].*$//'`
```
Comment (uberhacker) [inline tools/genpkgname.sh]: `CREW_LIB_PREFIX` constant?

File: tools/Makefile (hunk 1 above)
Comment (jam7) [reply tools/Makefile]: No.  We use /usr/local/lib/crew directory to place crew related stuff even on x86_64 currently.

File: tools/build.sh (hunk 2 above)
Comment (jam7) [reply tools/build.sh]: This is shell script.  How can I use `CREW_LIB_PREFIX`?

File: tools/genpkgname.sh (hunk 3 above)
Comment (jam7) [reply tools/genpkgname.sh]: This is shell script.  How can I use `CREW_LIB_PREFIX`?

File: tools/Makefile (hunk 1 above)
Comment (uberhacker) [reply tools/Makefile]: Fair enough.

File: tools/build.sh (hunk 2 above)
Comment (uberhacker) [reply tools/build.sh]: You can define a variable in shell with the same logic as ruby.  Different syntax but the logic to determine the directory is the same.

Comment (uberhacker) [reply tools/build.sh]: But in this case, since all the crew stuff goes into `/usr/local/lib/crew` anyway, it's pointless.

**Pull request merged**